### PR TITLE
Add tests code

### PR DESCRIPTION
**Pull request by archisman-dey**
This merge request adds the following features:
1. -c flag to stop after generating C code
2. -t flag to use stub functions
3. print() and println() while using -t

### CODE REVIEW

File: include/code_printer.h [hunk 1]
```diff
@@ -31,9 +31,9 @@
 #define _OPR_ASSIGNMENT " = " 
 
 #define _DT_INT_ "int"
-#define _DT_BOOL_ "bool"
```
Comment (VedantParanjape) [inline include/code_printer.h:34]: Don't remove this, let it be as it is.

File: src/ast.c [hunk 2]
```diff
@@ -431,6 +438,30 @@ ast_node_utility_function_call *create_send_rpmsg_call_node(ast_node_expression
     return utility_function_call;
 }
 
+ast_node_print_string_function_call *create_print_string_function_call_node(char *string, int add_newline)
+{
+    ast_node_print_string_function_call *print_function_call = (ast_node_print_string_function_call*)malloc(sizeof(ast_node_print_string_function_call));
+
+    print_function_call->node_type = AST_NODE_PRINT_STRING_FUNCTION_CALL;
+    print_function_call->add_newline = add_newline;
+
```
Comment (VedantParanjape) [inline src/ast.c:447]: Remove the extra blank line

File: src/ast.c [hunk 3]
```diff
@@ -431,6 +438,30 @@ ast_node_utility_function_call *create_send_rpmsg_call_node(ast_node_expression
...
+    return print_function_call;
+}
+
+ast_node_print_id_function_call *create_print_id_function_call_node(sym_ptr symbol_handle, int add_newline)
+{
+    ast_node_print_id_function_call *print_function_call = (ast_node_print_id_function_call*)malloc(sizeof(ast_node_print_id_function_call));
+
+    print_function_call->node_type = AST_NODE_PRINT_ID_FUNCTION_CALL;
+    print_function_call->add_newline = add_newline;
+
```
Comment (VedantParanjape) [inline src/ast.c:459]: Remove the extra blank line

File: src/lexer.l [hunk 4]
```diff
@@ -21,6 +21,7 @@ braces               ("{"|"}")
 parenthesis          ("("|")")
 integer              ([0-9]+)
 bool                 ("true"|"false")
+string               (\"(\\.|[^"\\])*\")
```
Comment (VedantParanjape) [inline src/lexer.l:24]: Please verify this on regex101.com, possibly share a prefilled link

**Review by VedantParanjape — CHANGES_REQUESTED** [review]

File: src/lexer.l (hunk 4 above)
Comment (archisman-dey) [reply src/lexer.l:24]: What this does is: start with a ", then match zero or more of (either (a backslash followed by a character) or (any character except " or backslash)), then end with a "
Added some unit tests here: https://regex101.com/r/6PgGC8/2

Comment (VedantParanjape) [reply src/lexer.l:24]: Cool, nice !

**Pull request merged**